Sources/SemanticIndex/SemanticIndexManager.swift

```diff
@@ -454,15 +466,7 @@ package final actor SemanticIndexManager {
         if !indexFilesWithUpToDateUnits, modifiedFilesIndex.hasUpToDateUnit(for: uri, mainFile: mainFile) {
           return nil
         }
-        return .headerFile(header: uri, mainFile: mainFile)
-      }
-      .filter {
-        switch inProgressIndexTasks[$0] {
-        case .waitingForPreparation, .creatingIndexTask:
-          return false
-        default:
-          return true
-        }
+        return (.headerFile(header: uri, mainFile: mainFile), modifiedFilesIndex.modificationDate(of: uri))
```

Comment:
If the main file has changed, should that also cause a re-index?

Reply:
Hmm seems like we don't currently consider that out of date, so I guess the answer is no

Reply:
Is that true? IIRC we actually do index the main file, not the header. Whether we should or not is another question, but that's how it works today.